quinn-proto/src/config.rs

```diff
@@ -601,27 +611,60 @@ where
 
 #[cfg(feature = "rustls")]
 impl ClientConfig<crypto::rustls::TlsSession> {
-    /// Add a trusted certificate authority
-    pub fn add_certificate_authority(
-        &mut self,
-        cert: Certificate,
-    ) -> Result<&mut Self, webpki::Error> {
-        let anchor = webpki::trust_anchor_util::cert_der_as_trust_anchor(&cert.inner.0)?;
-        Arc::make_mut(&mut self.crypto)
-            .root_store
-            .add_server_trust_anchors(&webpki::TLSServerTrustAnchors(&[anchor]));
-        Ok(self)
+    /// Create a client configuration that trusts the platform's native roots
+    #[cfg(feature = "native-certs")]
+    pub fn with_native_roots() -> Self {
+        let roots = match rustls_native_certs::load_native_certs() {
+            Ok(roots) => roots,
+            Err((Some(roots), e)) => {
+                tracing::warn!("couldn't load some default trust roots: {}", e);
+                roots
+            }
+            Err((None, e)) => {
+                tracing::warn!("couldn't load any default trust roots: {}", e);
+                rustls::RootCertStore::empty()
+            }
+        };
+
+        Self::new(roots, None)
+    }
+
+    /// Create a client configuration that trusts specified trust anchors
+    ///
+    /// If `ct_logs` is `None`, will use the default Certificate Transparency logs depending on
+    /// whether the `ct-logs` features is enabled or not. Otherwise, will use the specified logs.
```

Comment:
The feature is named "certificate-transparency". However, this is now convenient enough to pass in explicitly that I feel like maybe we should just drop the ct-logs dep entirely and let downstream users (who can much more easily update the package regularly) deal with it as needed.

Reply:
I think that makes sense, especially now that Rustls leaves certificate transparency implicitly disabled unless explicitly configured.